emulator.go

```diff
@@ -204,6 +210,44 @@ func (e *Emulator) Step() {
 	}
 }
 
+// Step performs one iteration of the waveform generation
+func (e *Emulator) Time_base_duration_loop(startTime int64, duration int64) []float64 {
+	var currentTime int64
+	var results []float64
+	for continue_flag := true; continue_flag; continue_flag = (currentTime - startTime) < duration {
+
+		currentTime = time.Now().Unix()
+		f := e.Fnom + e.Fdeviation
+
+		if e.fDeviationRemainingSamples > 0 {
+			e.fDeviationRemainingSamples--
+			if e.fDeviationRemainingSamples == 0 {
+				e.Fdeviation = 0.0
+			}
+		}
+
+		if e.V != nil {
+			e.V.stepThreePhase(e.r, f, e.Ts, e.SmpCnt)
+		}
+		if e.I != nil {
+			e.I.stepThreePhase(e.r, f, e.Ts, e.SmpCnt)
+		}
+		if e.T != nil {
+			e.T.stepTemperature(e.r, e.Ts)
+		}
+		if e.Sag != nil {
+			e.Sag.stepSag(e.r)
+		}
+
+		e.SmpCnt++
+		if int(e.SmpCnt) >= e.SamplingRate {
+			e.SmpCnt = 0
+		}
+		results = append(results, e.T.T)
+	}
+	return results
+}
```

Comment:
There is quite a fair bit of shared logic with `.Step()` in this function, so I think we should try and unify the implementation rather than having a seperate function. We might want to branch out another fn for a specific time-based task, but this is probably too similar to `.Step()`